### DISCUSSION TITLE
Help troubleshoot RLS SELECT returning null

### DISCUSSION THREAD

**Discussion by ptcampbell** [supabase/supabase#2374] [Questions] [answered]
I can successfully INSERT an item using RLS. 

But my SELECT query returns `null` despite the request containing a JWT which contains the right sub. 

I'm unsure how to go about troubleshooting the problem. When I disable RLS it returns the whole table which is to be expected. 

Anyone know what this might be? 

```sql
-- schema edited for brevity
create table public.boards (
  id            bigint generated by default as identity primary key,
  inserted_at   timestamp with time zone default timezone('utc'::text, now()) not null,
  name          text not null,
  created_by    uuid references public.users not null, -- which references the auth.users
  board_data    jsonb
);
comment on table public.boards is 'Collection of boards.';
alter table public.boards enable row level security;

-- policies
create policy "Allow individual read access" on public.boards for select using ( auth.uid() = created_by );
create policy "Allow individual insert access" on public.boards for insert with check ( auth.uid() = created_by );
```

**Comment by dshukertjr** [comment, marked as answer]
Hi @ptcampbell !

Your setup looks fine to me! Could you share the code you are using to retrieve the data from Supabase?

**Reply by ptcampbell** [reply]
@dshukertjr I managed to get it working. 

I narrowed it down to improper usage of `useSWR` with Next.js's API routes. 🤦

The fetcher function I passed into the useSWR hook was not right. Now I'm passing in the Supabase query which is being called from the component itself. 

Cheers.
